src/codeu/chat/server/Server.java

```diff
@@ -81,7 +81,30 @@ public Server(final Uuid id, final Secret secret, final Relay relay,final File l
     this.localFile = new LocalFile(new File(file.getPath()));//file path is given by user
     this.controller = new Controller(id, model,localFile);//Use the new constructor to create this new controller.
     this.relay = relay;
-
+    this.commands.put(NetworkCode.CONVERSATION_AUTHORITY_REQUEST, new Command()
+    {
+      @Override
+      public void onMessage(InputStream in, OutputStream out) throws IOException
+      {
+        final ConversationHeader conversation = ConversationHeader.SERIALIZER.read(in);
+        final User targetUser = User.SERIALIZER.read(in);
+        final User fromUser = User.SERIALIZER.read(in);
+        final String parameterString = Serializers.STRING.read(in);
+        if(model.isUser(conversation, fromUser.id) || model.isBannedUser(conversation, fromUser.id))
+        {
+          Serializers.INTEGER.write(out, NetworkCode.CONVERSATION_ACCESS_DENIED);
+        }
+        else if(model.isOwner(conversation, fromUser.id) && model.isCreator(conversation, targetUser.id))
```

Comment:
Personally, you could get a lot more done if you merge the else if and if into 
`if( model.isUser(conversation, fromUser.id) || model.isBannedUser(conversation, fromUser.id) || model.isCreator(conversation, targetUser.id))` because not even a Creator should be able to accidentally demote themselves.